nodejs/week3/meal/src/backend/api/meals.js

```diff
@@ -0,0 +1,209 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+const addMeal = async ({ body }) => {
+  const { title, description, maxNumberOfGuests, createdAt, price } = body;
+  return await knex("meals").insert({
+    title: title,
+    description: description,
+    maxNumberOfGuests: maxNumberOfGuests,
+    createdAt: createdAt,
+    price: price,
+  });
+};
+router.post("/", async (request, response) => {
+  addMeal({
+    body: request.body,
+  })
+    .then(result => response.json(result))
+    .catch(error => {
+      response.status(400).send("Bad request").end();
+      console.log(error);
+    });
+});
+//////
+const getMealsById = async ({ body, id }) => {
+  try {
+    const { title, description, maxNumberOfGuests, createdAt, price } = body;
+    return await knex("meals")
+      .where({
+        id: id,
+      })
+      .select("*");
+  } catch (error) {
+    console.log(error);
+  }
+};
+router.get("/:id", async (req, res) => {
+  getMealsById({
+    body: req.body,
+    id: req.params.id,
+  })
+    .then(result => res.json(result))
+    .catch(error => {
+      response.status(400).send("Bad request").end();
+      console.log(error);
+    });
+});
+//////
+const getUpdatedMeal = async ({ body, id }) => {
+  const { title, description, maxNumberOfGuests, createdAt, price } = body;
+  const meal = await knex.from("meals").select("*").where({
+    id: id,
+  });
+  if (meal.length === 0) {
+    throw new HttpError("Bad request", `Contact not found: ID ${id}!`, 404);
+  }
+  const queryDto = {
+    price: price,
+  };
+  if (Object.keys(queryDto).length !== 0) {
+    return await knex("meals")
+      .where({
+        id: id,
+      })
+      .update(queryDto);
+  } else return "Nothing updated!";
+};
+router.put("/:id", async (req, res) => {
+  getUpdatedMeal({
+    body: req.body,
+    id: req.params.id,
+  })
+    .then(result => res.json(result))
```

Comment:
Here you can also return something like 'Update was successful' if the update was indeed successful. But it's just an extra.